shared/stories/storybook.shared.js

```diff
@@ -18,8 +18,11 @@ const unexpected = (name: string) => () => {
  *                       The Provider will ignore all dispatched actions. It also wraps the component
  *                       tree in an <ErrorBoundary /> that adds auxiliary info in case of an error.
  */
+// Redux doens't allow swapping the store given a single provider so we use a new key to force a new provider to
+// work around this issue
+let uniqueProviderKey = 1
 const createPropProvider = (map: SelectorMap) => (story: () => React.Node) => (
-  <Provider store={createStore(state => state, map)}>
+  <Provider key={`provider:${uniqueProviderKey++}`} store={createStore(state => state, map)}>
```

Comment:
if you switch stories sometimes you'd get redboxes. Provider doesn't expect store to change dynamically and it has explicit errors about this, so as a workaround we entirely switch out the provider every time